buildrunner/docker/runner.py

```diff
@@ -251,6 +254,94 @@ def cleanup(self):
 
         self.container = None
 
+    def restore_caches(self, caches: OrderedDict, cache_archive_ext: str):  # pylint: disable=too-many-branches,too-many-locals
+        """
+        Restores caches from the host system to the destination location in the docker container.
+        """
+        if caches is None or not isinstance(caches, OrderedDict):
+            raise TypeError(f"Caches should be type OrderDict instead of {type(caches)}")
+
+        restored_cache_src = set()
+        for local_cache_archive_file, docker_path in caches.items():
+            if docker_path in restored_cache_src:
+                print(f"Destination path {docker_path} has already been matched and restored to the container "
+                      f"skipping {local_cache_archive_file} -> {docker_path}")
+                continue
+
+            # Check for prefix matching
+            if not os.path.exists(local_cache_archive_file):
+                cache_dir = os.path.dirname(local_cache_archive_file)
+
+                if not os.path.exists(cache_dir):
+                    print(f"Cache directory {cache_dir} does not exist, "
+                          f"skipping restore of archive {local_cache_archive_file}")
+                    continue
+
+                files = [f for f in listdir(cache_dir) if isfile(join(cache_dir, f))]
+
+                cache_key = local_cache_archive_file\
+                    .replace(f"{cache_dir}/", "")\
+                    .replace(f".{cache_archive_ext}", "")
+
+                most_recent_time = 0
+                local_cache_archive_match = None
+                for file in files:
+                    if file.startswith(cache_key):
+                        print(f"    starts with {cache_key}")
+                        curr_archive_file = join(cache_dir, file)
+                        mod_time = getmtime(curr_archive_file)
+                        if mod_time > most_recent_time:
+                            most_recent_time = mod_time
+                            local_cache_archive_match = curr_archive_file
+
+                if local_cache_archive_match is None:
+                    print(f"WARNING: Not able to restore cache {docker_path} since "
+                          f"there was no prefix matching for [{local_cache_archive_file}]")
```

Comment:
`no prefix match for ...` or `no matching prefix for ...`

Reply:
Should this be a WARNING? Or just a regular note?

Reply:
> `no prefix match for ...` or `no matching prefix for ...`


`no matching prefix for ...` reads better

- [ ] Update grammar to `no matching prefix for ...`

Reply:
> Should this be a WARNING? Or just a regular note?

Probably just a note, I guess it depends if you'd expect it to be matching or not.

- [ ] Remove `WARNING`